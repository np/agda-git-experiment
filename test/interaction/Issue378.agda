-- Andreas, 2012-11-06 this has been fixed with issue 655
module Issue378 where

infix  7 _≡_
infixl 9 _+_

postulate
  ℕ    : Set
  _≡_  : ℕ → ℕ → Set
  zero : ℕ
  _+_  : ℕ → ℕ → ℕ

postulate
  S₁ : ∀ {m n o} → m ≡ n → m ≡ o → n ≡ o
  S₅ : ∀ n → zero + n ≡ n

-- Calling Auto on the hole generates the invalid expression
-- S₁ (S₅ .n) (S₅ .n).
refl : ∀ {n} → n ≡ n
refl = {!-m!}
-- C-c C-a should work now
