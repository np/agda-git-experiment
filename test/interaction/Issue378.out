Agda2> (agda2-status-action "")
(agda2-info-action "*Type-checking*" "" nil)
(agda2-highlight-clear)
(agda2-info-action "*Type-checking*" "Checking Issue378 (Issue378.agda).\n" t)
(agda2-info-action "*Type-checking*" "Finished Issue378.\n" t)
(agda2-status-action "")
(agda2-info-action "*All Goals*" "?0 : {n : ℕ} → n ≡ n\n" nil)
((last . 1) . (agda2-goals-action '(0)))
Agda2> (agda2-give-action 0 "λ {n} → S₁ (S₅ n) (S₅ n)")
(agda2-status-action "")
(agda2-info-action "*All Goals*" "" nil)
((last . 1) . (agda2-goals-action '()))
Agda2> 